src/Aspire.Hosting/ResourceBuilderExtensions.cs

```diff
@@ -869,12 +870,6 @@ internal static IResourceBuilder<T> WithHttpHealthCheckInternal<T>(this IResourc
                 throw new DistributedApplicationException($"The endpoint '{endpointName}' on resource '{builder.Resource.Name}' was not using the '{desiredScheme}' scheme.");
             }
 
-            return Task.CompletedTask;
-        });
-
-        Uri? uri = null;
-        builder.ApplicationBuilder.Eventing.Subscribe<BeforeResourceStartedEvent>(builder.Resource, (@event, ct) =>
```

Comment:
BeforeResourceStartedEvent may fire before AfterEndpointsAllocatedEvent, so it makes sense to simply get the uri in the AfterEndpointsAllocatedEvent callback.